### PR TITLE
MFP: Additional viscosity model 

### PR DESCRIPTION
**Pull request by ThieJan**
Add viscosity model with dependency on temperature and concentration as presented in
Chierici (1994)

### CODE REVIEW

File: FEM/rf_mfp_new.cpp [hunk 1]
```diff
@@ -596,6 +597,16 @@ std::ios::pos_type CFluidProperties::Read(std::ifstream* mfp_file)
                 if (T_Process || fluid_id == 1)
                     viscosity_pcs_name_vector.push_back(arg2);
             }
+            // JM_visco
+            if (viscosity_model == 10)  // my(C,T),
+            {
+                in >> C_0;
+                in >> viscosity_T_shift;
```
Comment (wenqing) [inline FEM/rf_mfp_new.cpp]: This is not need. Temperature is changed to Kevin one during the computation.
There is a keyword in pcs: `$TEMPERATURE_UNIT` for input data. 
The default temperature unit for input is Celsius.

File: FEM/rf_mfp_new.cpp [hunk 2]
```diff
@@ -1877,7 +1888,24 @@ double CFluidProperties::Viscosity(double* variables)
...
+        // JM_visco
+        case 10:
+        {
+            if (!T_Process)
+                primary_variable[1] = T_0 + viscosity_T_shift;
+            else
+                primary_variable[1] +=
+                    viscosity_T_shift;  // JM if viscosity_T_shift==273 (user
+                                        // defined), Celsius can be used within
+                                        // this model
```
Comment (wenqing) [inline FEM/rf_mfp_new.cpp]: Not needed.

File: FEM/rf_mfp_new.h [hunk 3]
```diff
@@ -228,6 +228,7 @@ class CFluidProperties
     double viscosity;
     double viscosity0;
     double viscosity_T_star;
+    double viscosity_T_shift; // for viscosity model =10
```
Comment (wenqing) [inline FEM/rf_mfp_new.h]: Not needed.

**Comment by norihiro-w** [comment]
looks ok, except for the part of temperature shifting as Wenqing commented.

File: FEM/rf_mfp_new.cpp (hunk 1 above)
Comment (ThieJan) [reply FEM/rf_mfp_new.cpp]: Thanks for the comment. I will adapt this

Do I understand correctly that internally the temperature is always considered in Kelvin?

I rebased this changes from a quite old version of ogs5 we used for a project and think we received temperature values in celsius then as input for this function

Comment (wenqing) [reply FEM/rf_mfp_new.cpp]: You are right that the internal  internally the temperature is always considered in Kelvin. The Celsius can be used for input and output (see PR.  #115). By default, the temperature unit for input and output is Celsius.

If you have to read a temperature value, e.g a reference temperature, for a material model, you can do like the following
```
               in >> T_0;
                T_0 += TemperatureUnitOffset();
```
(see https://github.com/ufz/ogs5/blob/master/FEM/rf_mfp_new.cpp#L363-L364)

In your case, the input of `in >> viscosity_T_shift` is unnecessary.

Comment (ThieJan) [reply FEM/rf_mfp_new.cpp]: Thanks for referencing the PR. We hadn't noticed that change. I will adapt the changes accordingly

**Pull request merged**